From 69b0f72cb2df01e5f74eb23ed81007014b14f0dd Mon Sep 17 00:00:00 2001
From: Sergio Durigan Junior <sergiodj@chainguard.dev>
Date: Fri, 29 Aug 2025 20:06:13 -0400
Subject: [PATCH 09/11] Turn toolenvMicrosoft into a noop

We don't want to remove "systemcrypto" from GOEXPERIMENT, nor do we
make use of the ms_skipfipscheck tag.

Signed-off-by: Sergio Durigan Junior <sergiodj@chainguard.dev>
---
 src/cmd/dist/build.go | 44 ++-----------------------------------------
 1 file changed, 2 insertions(+), 42 deletions(-)

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index a3d6739a13..b9ce92a0a9 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -1423,48 +1423,8 @@ func updateEnv(name string, env []string, fn func(string) string) []string {
 
 // toolenvMicrosoft appends Microsoft-specific environment variables to env.
 func toolenvMicrosoft(env []string) []string {
-	// Keep all the experiments from the current
-	// environment, but with "systemcrypto" removed and
-	// "nosystemcrypto" added. This is used when building
-	// the Microsoft build of Go to favor portability.
-	// Go only uses crypto packages for non-crypto purposes,
-	// such as generating build IDs.
-	env = updateEnv("GOEXPERIMENT", env, func(value string) string {
-		if value == "" {
-			return "nosystemcrypto"
-		}
-		goexp := strings.Split(value, ",")
-		goexp = slices.DeleteFunc(goexp, func(s string) bool {
-			return s == "systemcrypto" || s == "nosystemcrypto" || s == "ms_nocgo_opensslcrypto"
-		})
-		goexp = append(goexp, "nosystemcrypto")
-		return strings.Join(goexp, ",")
-	})
-
-	// Build the Go toolchain with "-tags=ms_skipfipscheck". This
-	// allows toolchains not built with the systemcrypto goexperiment to be used
-	// when GODEBUG=fips140=on is set. For example, when running
-	// "GODEBUG=fips140=on go test ./..." or "GODEBUG=fips140=on go run .".
-	env = updateEnv("GOFLAGS", env, func(value string) string {
-		if value == "" {
-			return "-tags=ms_skipfipscheck"
-		}
-		goflags := strings.Fields(value)
-		// If -tags= is already set, append "ms_skipfipscheck" to it.
-		// If not, add it as a new flag.
-		var found bool
-		for i, e := range goflags {
-			if value, ok := strings.CutPrefix(e, "-tags="); ok {
-				goflags[i] = "-tags=" + value + ",ms_skipfipscheck"
-				found = true
-				break
-			}
-		}
-		if !found {
-			goflags = append(goflags, "-tags=ms_skipfipscheck")
-		}
-		return strings.Join(goflags, " ")
-	})
+	// Chainguard: we don't need to modify environment variables
+	// here.
 	return env
 }
 
-- 
2.51.0

